### ISSUE TITLE
Proposal : Use capital letter for quick access to micro-states

### ISSUE THREAD

**Issue by ralesi** [syl20bnr/spacemacs#3628]
For example, use `SPC W` and `SPC B` to enter micro-state the window and buffer modes.

**Comment by bmag** [comment]
I beg to differ, for two reasons. First, `SPC W` is already used by the eyebrowse layer. Second, we can potentially use `SPC B` (and other capital letters) as additional category prefixes in the future, which IMHO is more beneficial. Furthermore, `SPC w .` and `SPC b .` (and others?) already enter micro-states, so we already have quick access to micro-states on `SPC <letter> .`.

**Comment by StreakyCobra** [comment]
I agree with @bmag :-)

**Comment by ralesi** [comment]
Yeah,  I figured it was a no,  but did like having one less key for the micro state (actually same number if you count the shift) .  With the current setup though,  I wonder how often people enter the microstates.   Should we consider making some of the other keys enter the microstate as well?  Like `SPC b n`?

**Comment by robbyoconnor** [comment]
This would mean that we should switch perspectives back to `SPC L` if we went for this, since perspectives uses a microstate

**Comment by robbyoconnor** [comment]
I would prefer we not do this -- and I would also prefer we keep eyebrowse at `SPC W` -- I use eyebrowse occasionally.

**Comment by ralesi** [comment]
It seems this is a no, there are too many other prefixes used. One question, do we have the ability with the current micro state to enter it from another command. i.e SPC b n goes to next buffer as well as enter buffer micro state?

**Comment by a13ph** [comment]
@ralesi shift is a key too. So `SPC B` is same as `SPC b .`, or arguably even worse ergonomically, while being marginally faster in some cases.

Biggest issues of key layouts in Emacs are all the chording and overloading of weak fingers.

**Comment by bmag** [comment]
About using some keys to enter a micro-state, i.e. <kbd>SPB b n</kbd> to switch to next buffer and enter the buffer micro-state, I think that it is a good idea, but only in some cases.
For example, usually I want to use <kbd>SPC w [</kbd> and repeat it (or the `]`/`{`/`}` variants) a few times, so it would be easier to use if those commands entered the window micro-state. On the other hand, <kbd>SPC w j</kbd> (or `k`/`l`/`h`) I usually just want to use once and be done with it, so I wouldn't want them to enter the micro-state.
In short, I'm saying that it's a case-by-case decision.

**Comment by nixmaniack** [comment]
I was just thinking about this and thought why not just combine all microstates somewhere? I generally use microstates to perform some actions repeatedly (I guess that's why they are made of)
e.g.
`SPC M` then
`w` windows
`f` frames
`b` buffer
`l` layouts
`p` paste
`t` text scale
`T` transparency etc. 

I am thinking I would use `highlight-changes` and `diff-hl` microstates a lot for navigating buffer with changes. Also text/frame scale. So having such microstates under one umbrella would benefit a lot instead of guessing where they reside.

Any thoughts?

**Comment by StreakyCobra** [comment]
You can also use `SPC .` as prefix for this. It seems to be unused, and `.` it's already bind as microstate for some commands (`SPC b .`, `SPC w .`, …)

**Comment by nixmaniack** [comment]
@StreakyCobra Yes it can be. I meant "combine all microstates somewhere". But what's your opinion on "combining under"?

**Comment by StreakyCobra** [comment]
For me it doesn't matter, as I'm not using these micro-state that much. If you was thinking to move them, people will probably not like that. `SPC n .` is one keystroke less than `SPC . n .`, same for all directly entering micro-state. But I agree it could be nice to alias/copy all of them under a common prefix :-)

For me the ideal would be:
- Keeping traditional micro-states keybindings like `SPC b .`, `SPC w .`, …
- `SPC . [wfblptT…]` for entering micro-states from a central place.
- Adding a variable to `.spacemacs` that allows to specify all micro-states that should be entered automatically, like `SPC b n` entering the microstate directly.

**Comment by nixmaniack** [comment]
Current will remain as it is, only combine all of them under `SPC . [wfblptT…]` for ease of discovery at central place. It's same number of keystrokes `SPC . w` if you notice, I didn't mean `SPC . w .`

**Comment by StreakyCobra** [comment]
Yeah, for `SPC w .` it is not shorter, but for `SPC l` and `SPC . l` it is :wink: Anyway, I'm for this centralization. And special bonus if someone make a way to have a `.spacemacs` variable to define key-binding that should enter micro-state directly (like `SPC b n`).

**Comment by ralesi** [comment]
Is it even possible to add another prefix to start a microstate?  Also can we append to already defined microstate,  for example if we want to add some commands that require another package?

**Comment by StreakyCobra** [comment]
I'm not sure to understand exactly what you want, but maybe this: https://github.com/syl20bnr/spacemacs/blob/develop/doc/FAQ.org#change-or-define-an-alias-for-an-evil-leader-prefix (for the first question). For the second I don't know.

**Comment by ralesi** [comment]
Yes the alias would work for the first, for the second, I had wondered if it was possible to add additional commands to a micro-state within another packages' :init section

**Comment by ralesi** [comment]
So for example if there are commands  contingent on a package being loaded.

**Comment by ralesi** [comment]
New transient state allows this.
